pkgs/apps/vmd/binary.nix

```diff
@@ -0,0 +1,97 @@
+{ stdenv, lib, requireFile, makeWrapper, writeScriptBin, bash, perl, tcl-8_5, tk-8_5, symlinkJoin
+, netcdf, libGLU, xorg, fltk, vrpn, flex, bison, libGL_driver, cudatoolkit, autoPatchelfHook
+}:
+assert
+  lib.asserts.assertMsg
+  (stdenv.isLinux && stdenv.isx86_64)
+  "The VMD binaries with require an x86_64 linux OS with CUDA support.";
+
+let
+  homepage = "https://www.ks.uiuc.edu/Research/vmd/";
+  cudatoolkit_joined = symlinkJoin {
+    name = "cudatoolkit_joined";
+    paths = [ cudatoolkit.out cudatoolkit.lib ];
+  };
+
+in stdenv.mkDerivation rec {
+  pname = "vmd";
+  version = "1.9.3";
+
+  src = requireFile rec {
+    url = homepage;
+    name = "vmd-1.9.3.bin.LINUXAMD64-CUDA8-OptiX4-OSPRay111p1.opengl.tar.gz";
+    sha256 = "9427a7acb1c7809525f70f635bceeb7eff8e7574e7e3565d6f71f3d6ce405a71";
+  };
+
+  nativeBuildInputs = [
+    perl
+    makeWrapper
+    autoPatchelfHook
+  ];
+
+  buildInputs = [
+    libGLU
+    xorg.libX11
+    xorg.libXinerama
+    xorg.libXi
+    tcl-8_5
+    tk-8_5
+    netcdf
+    fltk
+    vrpn
+    flex
+    bison
+    libGL_driver
+    cudatoolkit_joined
+    # nvidia_x11
+  ];
+
+  postPatch = ''
+    substituteInPlace ./configure \
+      --replace '/usr/local' "$out" \
+      --replace '-ll' '-lfl'
+
+    patchShebangs ./configure
+  '';
+
+  sourceRoot = "vmd-${version}";
+
+  # non-standard configure script
+  configurePhase = ''
+    ./configure
+  '';
+
+  dontBuild = true;
+
+  preInstall = ''
+    cd src
+  '';
+
+  postInstall = ''
+    # Needs libcuda.so.1 but only finds libcuda.so
+    ln -s ${cudatoolkit}/targets/x86_64-linux/lib/stubs/libcuda.so $out/lib/libcuda.so.1
+
+    # Makes tachyon available
+    ln -s $out/lib/vmd/{stride,surf,tachyon}_LINUXAMD64 $out/bin/.
+  '';
+
+  # libnvcuvid.so depends on the linuxPackages.nvidia_x11.
+  # This might be overriden in configuration.nix and should be detected at runtime at
+  # /var/run/opengl-driver/lib
+  autoPatchelfIgnoreMissingDeps = true;
+
+  postFixup = ''
+    wrapProgram $out/bin/vmd \
+      --set "LC_ALL" "C"
```

Comment:
What is that required for? Should we really override local LC settings?

Reply:
VMD tends to use system locales for its parser. In consequence, if you are on a system with a locale, that uses comma as decimal separator (like german aka it's a problem specifically for me :smile: ), the VMD parsers will fail to parse anything, as they suddenly expect xyz, molden, cube, pdb, ... files to use comma as a separator.

Reply:
OK, I see.